docs/tutorial/dark-mode.md

```diff
@@ -80,65 +121,61 @@ Add the following lines to the `index.html` file:
 
-To add listeners and handlers, add the following lines to the `renderer.js` file:
+> See it in action! Load the example using Electron Fiddle and then click the "Toggle Dark Mode" button.
 
-```javascript
-const { ipcRenderer } = require('electron')
+The `renderer.js` file is responsible for controlling the `<button>` functionality.
 
+```js title='renderer.js'
 document.getElementById('toggle-dark-mode').addEventListener('click', async () => {
-  const isDarkMode = await ipcRenderer.invoke('dark-mode:toggle')
+  const isDarkMode = await window.darkMode.toggle()
   document.getElementById('theme-source').innerHTML = isDarkMode ? 'Dark' : 'Light'
 })
 
 document.getElementById('reset-to-system').addEventListener('click', async () => {
-  await ipcRenderer.invoke('dark-mode:system')
+  await window.darkMode.system()
   document.getElementById('theme-source').innerHTML = 'System'
 })
 ```
 
-If you run your code at this point, you'll see that your buttons don't do
-anything just yet, and your Main process will output an error like this when
-you click on your buttons:
-`Error occurred in handler for 'dark-mode:toggle': No handler registered for 'dark-mode:toggle'`
-This is expected — we haven't actually touched any `nativeTheme` code yet.
+It adds `'click'` [event listeners](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener) to each button. The functions make calls to the `window.darkMode` API. The `darkMode` API is loaded onto the `window` object by the `preload.js` script.
+
+```js title='preload.js'
+const { contextBridge, ipcRenderer } = require('electron')
 
-Now that we're done wiring the IPC from the Renderer's side, the next step
-is to update the `main.js` file to handle events from the Renderer process.
+contextBridge.exposeInMainWorld('darkMode', {
+  toggle: () => ipcRenderer.invoke('dark-mode:toggle'),
+  system: () => ipcRenderer.invoke('dark-mode:system')
+})
+```
 
-Depending on the received event, we update the
-[`nativeTheme.themeSource`](../api/native-theme.md#nativethemethemesource)
-property to apply the desired theme on the system's native UI elements
-(e.g. context menus) and propagate the preferred color scheme to the Renderer
-process:
+The `'dark-mode:toggle'` and `'dark-mode:system'` strings correspond to IPC Channel names that the Main Process can use to perform the necessary mutations to the `nativeTheme` object. The use of `contextBridge` is important for [Electron application security best practices](./context-isolation.md).
 
-* Upon receiving `dark-mode:toggle`, we check if the dark theme is currently
-active using the `nativeTheme.shouldUseDarkColors` property, and set the
-`themeSource` to the opposite theme.
-* Upon receiving `dark-mode:system`, we reset the `themeSource` to `system`.
+Finally, the `main.js` file represents the Main Process and contains the actual `nativeTheme` API.
```

Comment:
nit: we're not very consistent all around with the capitalization of "Main Process". I usually just say `main process` all-lowercase. Thoughts? 🤔